app/src/main/java/com/example/concordia_campus_guide/Fragments/PathInfoCardFragment/PathInfoCardFragment.java

```diff
@@ -0,0 +1,272 @@
+
+        for (int i = 0; i < walkingPointList.size() - 1; i++) {
+            WalkingPoint startWalkingPoint = walkingPointList.get(i);
+            WalkingPoint endWalkingPoint = walkingPointList.get(i + 1);
+            boolean last_one = i == directionList.size() - 1;
+            setIndoorComponents(startWalkingPoint, endWalkingPoint, layout, layoutParams, last_one);
+        }
+
+        this.totalDuration = (totalDistance * 60.0 / 5.0);
+        createImageButton(layout, layoutParams, R.drawable.ic_directions_walk_white);
+        setRecyclerView();
+    }
+
+    public void setIndoorComponents(WalkingPoint startWalkingPoint, WalkingPoint endWalkingPoint, LinearLayout layout, LinearLayout.LayoutParams layoutParams, boolean lastOne) {
+        distanceBetweenPoints += pathInfoCardViewModel.getDistanceFromLatLonInKm(startWalkingPoint.getCoordinate().getLatitude(), startWalkingPoint.getCoordinate().getLongitude(), endWalkingPoint.getCoordinate().getLatitude(), startWalkingPoint.getCoordinate().getLongitude());
+        totalDistance += distanceBetweenPoints;
+        // on average a person walks 5km / hr
+        long timeTakenInMinutes;
+        String description = "";
+        if (startWalkingPoint.getPointType() == PointType.CLASSROOM) {
+            description = "Leave classroom";
+            timeTakenInMinutes = (long) (distanceBetweenPoints * 60 / 5);
+            addIndoorDirection(startWalkingPoint.getCoordinate().getLatLng(), endWalkingPoint.getCoordinate().getLatLng(), description, distanceBetweenPoints, timeTakenInMinutes);
+            distanceBetweenPoints = 0;
+        }
+        PointType pt = endWalkingPoint.getPointType();
+        switch (pt) {
+            case ELEVATOR:
+                if (startWalkingPoint.getPointType() != pt.ELEVATOR) {
+                    description = "Walk towards elevator";
+                    timeTakenInMinutes = (long) (distanceBetweenPoints * 60 / 5);
+                    addIndoorDirection(startWalkingPoint.getCoordinate().getLatLng(), endWalkingPoint.getCoordinate().getLatLng(), description, distanceBetweenPoints, timeTakenInMinutes);
+                    distanceBetweenPoints = 0;
+                    createImageButton(layout, layoutParams, R.drawable.ic_elevator_white);
+                    if (!lastOne)
+                        setDividerTextView(layout, layoutParams);
+                }
+                break;
+            case ENTRANCE:
+                description = "Enter building";
+                timeTakenInMinutes = (long) (distanceBetweenPoints * 60 / 5);
+                addIndoorDirection(startWalkingPoint.getCoordinate().getLatLng(), endWalkingPoint.getCoordinate().getLatLng(), description, distanceBetweenPoints, timeTakenInMinutes);
+                distanceBetweenPoints = 0;
+                break;
+            case STAFF_ELEVATOR:
```

Comment:
This methods is too complicated to be understood. Can you break down to smaller methods ?